diff --git a/log/log.go b/log/log.go
index 3a1c653..16e45ea 100644
--- a/log/log.go
+++ b/log/log.go
@@ -48,7 +48,7 @@ func New(options Options) (Factory, error) {
 		DisableColors:    logOptions.DisableColor || logFilePath != "",
 		DisableTimestamp: !logOptions.Timestamp && logFilePath != "",
 		FullTimestamp:    logOptions.Timestamp,
-		TimestampFormat:  "-0700 2006-01-02 15:04:05",
+		TimestampFormat:  "[2006-01-02 15:04:05 UTC-07]",
 	}
 	factory := NewDefaultFactory(
 		options.Context,
