diff --git a/option/options.go b/option/options.go
index fcca94c..222a418 100644
--- a/option/options.go
+++ b/option/options.go
@@ -43,7 +43,7 @@ type LogOptions struct {
 	Level        string `json:"level,omitempty"`
 	Output       string `json:"output,omitempty"`
 	Timestamp    bool   `json:"timestamp,omitempty"`
-	DisableColor bool   `json:"-"`
+	DisableColor bool   `json:"disable_color,omitempty"`
 }
 
 type StubOptions struct{}
